modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/HashIndex.java

```diff
@@ -55,4 +57,18 @@ HashIndexStorage getOrCreateStorage(RocksDbMvPartitionStorage partitionStorage)
     void destroy() {
         storages.forEach((partitionId, storage) -> storage.destroy());
     }
+
+    /**
+     * Deletes the data associated with the partition in the index, using passed write batch for the operation.
+     *
+     * @throws RocksDBException If failed to delete data.
+     */
+    void destroy(int partitionId, WriteBatch writeBatch) throws RocksDBException {
```

Comment:
Please point out that we don't just delete the contents of the index, but also close it if it exists.